src/core/modules/Planet.cpp

```diff
@@ -1275,34 +1379,53 @@ QString Planet::getInfoStringExtra(const StelCore *core, const InfoStringGroup&
 				double raDiff = StelUtils::fmodpos((raMoon - raSun)/M_PI_180, 360.0);
 				if (raDiff < 3. || raDiff > 357.)
 				{
-					SolarEclipse solarEclipse;
-					Vec3d pos = solarEclipse.point(core1);
+					double JD = core1->getJD();
+					double dRatio,latDeg,lngDeg,altitude,pathWidth,duration,magnitude;
+					SolarEclipseData(JD,dRatio,latDeg,lngDeg,altitude,pathWidth,duration,magnitude);
 
-					if (pos[0] < 90.) // only display when shadow axis is touching Earth
+					if (pathWidth > 0.) // only display when shadow axis is touching Earth
 					{
+						oss << QString("%1: %2 ").arg(
+								 q_("Moon/Sun diameter ratio"), // It seems magnitude of total/annular eclipses sometimes represented by this value
+								 QString::number(dRatio, 'f', 3));
+						if (dRatio < 1.0)
+							oss << QString(qc_("(annular)","type of solar eclipse"));
+						else
+							oss << QString(qc_("(total)","type of solar eclipse"));
+						oss << "<br/>";
+						double centralDuraton = abs(duration);
+						int durationMinute = int(centralDuraton);
+						int durationSecond = round((centralDuraton - durationMinute) * 60.);
+						if (durationSecond>59)
+						{
+							durationMinute += 1;
+							durationSecond = 0;
+						}
+						oss << QString("%1: %2%3 %4%5<br/>").arg(
+								 q_("Central eclipse duration"),
+								 QString::number(durationMinute),
+								 q_("m"),
+								 QString::number(durationSecond),
+								 q_("s"));
 						QString info = q_("Center of solar eclipse (Lat./Long.)");
 						if (withDecimalDegree)
-							oss << QString("%1: %2°/%3°<br />").arg(info).arg(pos[0], 5, 'f', 4).arg(pos[1], 5, 'f', 4);
+							oss << QString("%1: %2°/%3°<br />").arg(info).arg(latDeg, 5, 'f', 4).arg(lngDeg, 5, 'f', 4);
 						else
-							oss << QString("%1: %2/%3<br />").arg(info, StelUtils::decDegToDmsStr(pos[0]), StelUtils::decDegToDmsStr(pos[1]));
+							oss << QString("%1: %2/%3<br />").arg(info, StelUtils::decDegToDmsStr(latDeg), StelUtils::decDegToDmsStr(lngDeg));
 						StelLocation loc = core->getCurrentLocation();
 						// distance between center point and current location
-						double distance = loc.distanceKm(pos[1], pos[0]);
-						double azimuth = loc.getAzimuthForLocation(pos[1], pos[0]);
+						double distance = loc.distanceKm(lngDeg, latDeg);
+						double azimuth = loc.getAzimuthForLocation(lngDeg, latDeg);
 
 						oss << QString("%1 %2 %3 %4°<br/>").arg(
 								 q_("Shadow center point is"),
 								 QString::number(distance, 'f', 1),
 								 q_("km towards azimuth"),
 								 QString::number(azimuth, 'f', 1));
-						oss << QString("%1: %2 ").arg(
-								 q_("Magnitude of central eclipse"),
-								 QString::number(pos[2], 'f', 3));
-						if (pos[2] < 1.0)
-							oss << QString(qc_("(annular)","type of solar eclipse"));
-						else
-							oss << QString(qc_("(total)","type of solar eclipse"));
-						oss << "<br/>";
+						oss << QString("%1: %2 %3<br/>").arg(
+								 q_("Width of umbra/antumbra"),
```

Comment:
Can you separate and give two separate words: either "umbra" (for total eclipses or total phase of hybrid) or "antumbra" (annular/annular phase of hybrid)?